### ISSUE TITLE
config files for conditional training on segmentation maps

### ISSUE THREAD

**Issue by ash80** [CompVis/taming-transformers#17]
Great paper! I am trying to retrain this model on an image dataset where I'm able to generate the segmentation masks using DeepLab v2. However, I don't have a config yaml file for training transformer as for faceHQ or D-RIN. Could you please provide a sample yaml file training with segmentation masks? Many Thanks

**Comment by ink1** [comment]
I'm also interested in that, see #16 
The best starting I could find is the yaml file shared with sflckr checkpoint. Replace validation by train at the end of the file.
But my progress basically stops there.

**Comment by attashe** [comment]
I am trying to train this on Flicklr-30k dataset and after 21 epochs intermediate results nothing changed. Config from sflckr checkpoint.
![Снимок экрана 2021-02-04 в 19 38 57](https://user-images.githubusercontent.com/8243605/106924964-ae8ca780-6720-11eb-8bcf-f489992b2a4e.png)
![Снимок экрана 2021-02-04 в 19 39 58](https://user-images.githubusercontent.com/8243605/106925119-d67c0b00-6720-11eb-9fa8-b28c8ede3cc7.png)
![Снимок экрана 2021-02-04 в 19 40 07](https://user-images.githubusercontent.com/8243605/106925124-d714a180-6720-11eb-8872-70f4698bdbdc.png)

**Comment by akmtn** [comment]
Distributed sflckr.yaml seems insufficient for training, because some settings are lacking. For example, models.params.lossconfig

Hi, authors, 
Could you please provide a sample yaml file training with segmentation masks? Thanks.

**Comment by pesser** [comment]
Added config and loss to train cond stage on segmentation maps (`configs/coco_cond_stage.yaml` and `configs/sflckr_cond_stage.yaml`). Optionally, you can also extract weights of cond stage from the transformer checkpoints,
```
python scripts/extract_submodel.py logs/2021-01-20T16-04-20_coco_transformer/checkpoints/last.ckpt coco_cond_stage.ckpt cond_stage_model
```
and fine-tune from there (maybe adjust data section of config):
```
python main.py --base configs/coco_cond_stage.yaml -t True --gpus 0, model.params.ckpt_path=coco_cond_stage.ckpt
```

**Comment by kampta** [comment]
I have a related question.  It looks like the current config file `sflckr_cond_stage.yaml` leads to resizing the image to SmallestMaxSize=256. So the model was essentially trained using smaller (resized) images. Is the model checkpoint provided trained using the same config? I'd imagine in order to sample high res images, we need to just crop the images without resizing.

**Comment by ali-design** [comment]
Thank you for the great effort! 
How can I train the conditional transformer when I would like to condition image on a vector (as opposed to depth map or class label)? Specifically how the config would look like?

**Comment by kaihe** [comment]
> 

@kampta any luck in reproducing sflckr result **without** resizing ? @pesser would you please give some comments?

**Comment by kaihe** [comment]
> I have a related question. It looks like the current config file `sflckr_cond_stage.yaml` leads to resizing the image to SmallestMaxSize=256. So the model was essentially trained using smaller (resized) images. Is the model checkpoint provided trained using the same config? I'd imagine in order to sample high res images, we need to just crop the images without resizing.

according SmallestMaxSize is to upscale instead of downscale image
![image](https://user-images.githubusercontent.com/7460776/150935540-3a80fadd-2955-42ad-b2bb-d0cfba8ab166.png)

**Comment by Kai-0515** [comment]
> Added config and loss to train cond stage on segmentation maps (`configs/coco_cond_stage.yaml` and `configs/sflckr_cond_stage.yaml`). Optionally, you can also extract weights of cond stage from the transformer checkpoints,
> 
> ```
> python scripts/extract_submodel.py logs/2021-01-20T16-04-20_coco_transformer/checkpoints/last.ckpt coco_cond_stage.ckpt cond_stage_model
> ```
> 
> and fine-tune from there (maybe adjust data section of config):
> 
> ```
> python main.py --base configs/coco_cond_stage.yaml -t True --gpus 0, model.params.ckpt_path=coco_cond_stage.ckpt
> ```

hello How can I sample from a segmentation model? segmentation model can't sample as readme, because VQSegmentationModel has no attribute 'encode_to_z' but make_samples.py use it
 looking forward to your reply

**Comment by Kai-0515** [comment]
> I am trying to train this on Flicklr-30k dataset and after 21 epochs intermediate results nothing changed. Config from sflckr checkpoint. ![Снимок экрана 2021-02-04 в 19 38 57](https://user-images.githubusercontent.com/8243605/106924964-ae8ca780-6720-11eb-8bcf-f489992b2a4e.png) ![Снимок экрана 2021-02-04 в 19 39 58](https://user-images.githubusercontent.com/8243605/106925119-d67c0b00-6720-11eb-9fa8-b28c8ede3cc7.png) ![Снимок экрана 2021-02-04 в 19 40 07](https://user-images.githubusercontent.com/8243605/106925124-d714a180-6720-11eb-8872-70f4698bdbdc.png)

hello  could please tell me how u sample from segmentation model. I sample as the readme failed, because VQSegmentationModel has no attribute 'encode_to_z'
